flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/rules/datastream/DataStreamMatchRule.scala

```diff
@@ -42,11 +42,13 @@ class DataStreamMatchRule
       RelOptRule.convert(logicalMatch.getInput, FlinkConventions.DATASTREAM)
 
     try {
-      Class.forName("org.apache.flink.cep.pattern.Pattern")
+      Class
+        .forName("org.apache.flink.cep.pattern.Pattern",
+          false,
+          Thread.currentThread().getContextClassLoader)
     } catch {
       case ex: ClassNotFoundException => throw new TableException(
-        "MATCH RECOGNIZE clause requires flink-cep dependency to be present on the classpath.",
-        ex)
+        "MATCH RECOGNIZE clause requires flink-cep dependency to be present on the classpath.")
```

Comment:
Shouldn't we at least log to original exception?

Reply:
The thing is we shouldn't, because:
- sql-client prints only the root cause
- we already know what is the original exception. There is only one possible exception there.

Reply:
re 2: We know what has happened but I think that `ClassNotFoundException` can have multiple different causes. One of the is that class was indeed not found, but I think it can also be thrown if class was found, but JVM was failed to load/initialize it. Hiding this kind of things might make debugging/troubleshooting impossible/more difficult.  
re 1: if that's the case, I would be inclined to change it at some point of time. Maybe log causes in sql client to a file or provide `verbose` mode. I'm not saying to do it now, but decision to ignore/hide the exception cause shouldn't be made at this level anyway.

Reply:
Agreed, fixed.